src/meta.jl

```diff
@@ -88,24 +115,16 @@ end
 becomes `while_loop((i,loop_sum)->i<10, (i,loop_sum)->[i+1, loop_sum+i], [i, loop_sum])`.
 """
 macro tf(ex)
-    is_assign(arg::Expr) = arg.head == Symbol("=")
-    is_assign(arg) = false
-    if ex.head == :block
-        res = Expr(:block)
-        for arg in ex.args
-            if is_assign(arg)
-                push!(res.args, _tf(arg))
+    if ex.head == :while
+        tf_while(ex)
+    else
+        MacroTools.prewalk(ex) do x
```

Comment:
Will do. It is basically doing a recursive find and replace, with the replacement also being searched within. I think. (This is the bit I stole from @MikeInnes )
See also https://github.com/MikeInnes/MacroTools.jl/issues/34